source/dhtslib/faidx.d

```diff
@@ -109,69 +101,46 @@ struct IndexedFastaFile {
 
     /// Fetch sequence in region by assoc array-style lookup:
     /// `string sequence = fafile["chr2:20123-30456"]`
-    auto opIndex(string region)
+    auto opIndex(CoordSystem cs)(ChromCoordinates!cs region)
     {
         return fetchSequence(region);
     }
 
-    /// Fetch sequence in region by assoc array-style lookup:
-    /// `string sequence = fafile.fetchSequence("chr2:20123-30456")`
-    string fetchSequence(string region)
```

Comment:
I feel conflicted about this. I feel letting anyone bypass the coordinate system will only invite headaches and encourage people to not use it. If we provide easy overloads, easy access into the coordinate system, and extensive documentation there is no reason we should allow people to bypass it. If someone really wants to use int coordinates directly, use a coordinate overload (Zbho, Obho ...). At least then it is clearly defined what their coordinate space is. Also APIs change with major version  releases, which this will be included in. Deprecation is okay for subversion changes but I don't know if we should do that here.

Reply:
OK , you convinced me

Might as well go all-in

In this case need to make it as ergonomic as possible tho